docs/design/kubeadm-api.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+type MasterConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Data shared between phases. Used for defaulting
+	Master     Master     `json:"master"`
+	Networking Networking `json:"networking"`
+	CertificatesDir string `json:"certificatesDir"`
+
+	// Phases
+	Certificates     Certificates     `json:"certificates"`
```

Comment:
`Certificates` is not clear about its intents. If we have certificates then why do we need `CertificatesDir`? And while I may understand the design, the API itself doesn't seem self-explanatory and I believe will create confusion.